### PR TITLE
Settings UI

### PR DESCRIPTION
**Pull request by mikolajmirko**
Need logic for:

- Language overwrite change
- Profile name edit
- Account password change
- Account closing

TODO later:

- Change link to term of service and version (when ready)

### CODE REVIEW

**Comment by mikolajmirko** [comment]
Ready for review

File: i18n/en_US.json [hunk 1]
```diff
@@ -446,6 +481,7 @@
 		"everyday": "Codziennie"
 	},
 	"alert": {
+		"errorOccured": "Error occured",
```
Comment (stasgora) [inline i18n/en_US.json:484]: occu**rr**ed x2

File: i18n/en_US.json [hunk 2]
```diff
@@ -70,6 +72,39 @@
...
+					"changeLanguageLabel": "App language",
+					"languages": {
+						"default": "System (default)",
+						"pl": "Polish",
+						"en": "English"
+					},
+					"showAppInfoLabel": "About the app",
+					"appInfo": {
+						"version": "Version",
+						"description": "Fokus is a mobile application that helps kids complete their daily tasks under parental control. The app was created as part of a group project at Gdansk University of Technology.",
```
Comment (stasgora) [inline i18n/en_US.json:99]:
```suggestion
						"description": "Fokus is a mobile application that helps kids complete their daily tasks under parental control. It was created as part of a team project at Gdansk University of Technology.",
```

File: i18n/pl_PL.json [hunk 3]
```diff
@@ -70,6 +72,39 @@
...
+					"changeLanguageLabel": "Język aplikacji",
+					"languages": {
+						"default": "Systemowy (domyślny)",
+						"pl": "Polski",
+						"en": "Angielski"
+					},
+					"showAppInfoLabel": "O aplikacji",
+					"appInfo": {
+						"version": "Wersja",
+						"description": "Fokus to aplikacja mobilna pomagająca dzieciom i ich opiekunom w realizacji codziennych zadań. Aplikacja powstała w ramach projektu grupowego realizowanego na Politechnice Gdańskiej.",
```
Comment (stasgora) [inline i18n/pl_PL.json:99]:
```suggestion
						"description": "Fokus to aplikacja mobilna pomagająca dzieciom w realizacji codziennych zadań pod kontrolą opiekunów. Powstała w ramach projektu grupowego realizowanego na Politechnice Gdańskiej.",
```

File: lib/widgets/dialogs/app_info_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import 'package:fokus/utils/dialog_utils.dart';
+import 'package:fokus/utils/theme_config.dart';
+import 'package:fokus/widgets/dialogs/general_dialog.dart';
+import 'package:lottie/lottie.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class AppInfoDialog extends StatelessWidget {
+	final String _settingsKey = 'page.settings.content.appSettings.appInfo';
+
+	final String version = '0.0.1';
```
Comment (stasgora) [inline lib/widgets/dialogs/app_info_dialog.dart:12]: Use [package_info](https://pub.dev/packages/package_info)

File: lib/utils/dialog_utils.dart [hunk 5]
```diff
@@ -45,36 +47,57 @@ void showHelpDialog(BuildContext context, String helpPage) {
...
-									Navigator.pop(c, true);
-								else {
-									Navigator.of(context).pop();
-									Navigator.of(context).pop();
-								}
+Future<bool> showExitFormDialog(BuildContext context, bool isSystemPop, bool isDataChanged) {
+	if (!isDataChanged) {
+		Navigator.pop(context, true);
+		return Future.value(false);
+	} else {
```
Comment (stasgora) [inline lib/utils/dialog_utils.dart:54]: Don't really need the _else_ since you are returning in the _if_ :wink:

**Review by stasgora — CHANGES_REQUESTED** [review]

File: lib/widgets/dialogs/app_info_dialog.dart (hunk 4 above)
Comment (mikolajmirko) [reply lib/widgets/dialogs/app_info_dialog.dart]: obraz

**Review by stasgora — APPROVED** [review]

**Pull request merged**